### PR TITLE
[Merged by Bors] - TY-2315 filter previously seen articles

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**

- [TY-2315]

**Summary**

- update `ArticleFilter` trait with `history: &[HistoricDocument]`
- impl `HistoryFilter` and apply it in `CommonFilter`

[TY-2315]: https://xainag.atlassian.net/browse/TY-2315?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine_core/core/src/document.rs [hunk 1]
```diff
@@ -282,6 +282,19 @@ mod tests {
         }
     }
 
+    impl TryFrom<Article> for HistoricDocument {
```
Comment (acrrd) [inline discovery_engine_core/core/src/document.rs:285]: If we need  it make tests simpler we can define it in test module instead.

File: discovery_engine_core/core/src/stack/filters.rs [hunk 2]
```diff
@@ -12,23 +12,51 @@
...
+    fn apply(
+        history: &[HistoricDocument],
+        stack: &[Document],
+        articles: Vec<Article>,
+    ) -> Result<Vec<Article>, GenericError>;
+}
+
+struct HistoryFilter;
+
+impl ArticleFilter for HistoryFilter {
```
Comment (acrrd) [inline discovery_engine_core/core/src/stack/filters.rs:35]: Can't we do this already in `Duplicatefilter`?

File: discovery_engine_core/core/src/document.rs (hunk 1 above)
Comment (janpetschexain) [reply discovery_engine_core/core/src/document.rs:285]: not sure what you mean? this impl is already in a test module.

File: discovery_engine_core/core/src/stack/filters.rs (hunk 2 above)
Comment (janpetschexain) [reply discovery_engine_core/core/src/stack/filters.rs:35]: sure we can. my impression of the previous design was that all filters should be separate such that they can be combined in the `CommonFilter` later on. but we can also combine the `HistoryFilter` and the `DuplicateFilter` into one, i don't mind much.

File: discovery_engine_core/core/src/document.rs (hunk 1 above)
Comment (acrrd) [reply discovery_engine_core/core/src/document.rs:286]: 🙈 I'm blind sorry.

**Review by acrrd — APPROVED** [review]

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**